client/asset/btc/btc.go

```diff
@@ -3507,7 +3508,18 @@ func (btc *baseWallet) DepositAddress() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	return btc.stringAddr(addr, btc.chainParams)
+	addrStr, err := btc.stringAddr(addr, btc.chainParams)
+	if err != nil {
+		return "", err
+	}
+	// Be extra cautious and ensure the wallet gave us an address for which we
+	// can retrieve the private keys, regardless of what ownsAddress would say.
+	priv, err := btc.node.privKeyForAddress(addrStr)
```

Comment:
Gave me no trouble in testing, but if `DepositAddress` is used while the wallet is locked, this will fail.

I think I'll make this check run only if the wallet is unlocked (e.g. when placing an order or making a swap or refund).  Either that or remove the check entirely.